File: typescript/questions/canJump.ts
function canJump(nums: number[]): boolean {
  let nl = nums.length;
  let mp = 0;
  for (let i = 0; i <= mp; ++i) {
    mp = Math.max(mp, i + nums[i]);
    if (mp >= nl - 1) {
      return true;
    }
  }
  return false;
}

File: rust/questions/can_jump.rs
impl Solution {
    pub fn can_jump(nums: Vec<i32>) -> bool {
        let nl = nums.len();
        let mut mp: usize = 0;
        let mut p: usize = 0;
        while p <= mp {
            mp = mp.max(p + nums[p] as usize);
            if mp >= nl - 1 {
                return true;
            }
            p += 1;
        }
        return false;
    }
}
